python/python/bystro/proteomics/annotation_interface.py

```diff
@@ -0,0 +1,188 @@
+"""Query an annotation file and return a list of sample_ids and genes meeting the query criteria."""
+import logging
+import math
+from dataclasses import dataclass
+from typing import Any
+
+import numpy as np
+import pandas as pd
+from bystro.utils.config import get_opensearch_config
+from opensearchpy import OpenSearch
+
+logger = logging.getLogger(__file__)
+
+OPENSEARCH_CONFIG = get_opensearch_config()
+HETEROZYGOTE_DOSAGE = 1
+HOMOZYGOTE_DOSAGE = 2
+MISSING_GENO_DOSAGE = np.nan
+ONE_DAY = "1d"  # default keep_alive time for opensearch point in time index
+
+# The fields to return for each variant matched by the query
+OUTPUT_FIELDS = ["refSeq.name2", "homozygotes", "heterozygotes", "missingGenos"]
+
+
+@dataclass
+class OpenSearchQueryConfig:
```

Comment:
I would keep this as a Struct. We are using 3 different slot class libraries, let's stick to 1.